### PR TITLE
Add Episodic RL Http API interface

### PR DESCRIPTION
**Pull request by homezcx**
This PR adds the Episodic RL interface for HTTP_API_HOST. Like the existing EPISODE_EH_* implementation, you will need to specify below:

```
        "http.api.key": "{key}",
	"episode.http.api.host": "{endpoint}",
	"interaction.http.api.host": "{endpoint}",
	"observation.http.api.host": "{endpoint}",
	"episode.sender.implementation": "EPISODE_HTTP_API_SENDER",
	"interaction.sender.implementation": "INTERACTION_HTTP_API_SENDER",
	"observation.sender.implementation": "OBSERVATION_HTTP_API_SENDER",
```

### CODE REVIEW

**Review by jackgerrits — APPROVED** [review]

**Comment by rajan-chari** [comment]
I have a quick question.  This seems to be defining new constants but using them in the same way as before.  Is there a reason to not use the old constants ... in terms of end points etc?

**Review by rajan-chari — CHANGES_REQUESTED** [review]
I have a quick question. This seems to be defining new constants but using them in the same way as before. Is there a reason to not use the old constants ... in terms of end points etc?

**Comment by homezcx** [comment]
> I have a quick question. This seems to be defining new constants but using them in the same way as before. Is there a reason to not use the old constants ... in terms of end points etc?

Actually, no specific reason for me not to use the old constants.  I was mainly following the existing pattern put in Episode_EH_* to have EPISODE_APIM_*. If you think we want to clean up in this PR, I can do a follow-up patch too.

**Comment by rajan-chari** [comment]
> > I have a quick question. This seems to be defining new constants but using them in the same way as before. Is there a reason to not use the old constants ... in terms of end points etc?

> Actually, no specific reason for me not to use the old constants.  I was mainly following the existing pattern put in Episode_EH_* to have EPISODE_APIM_*. If you think we want to clean up in this PR, I can do a follow-up patch too. 

Ah ok.  I see your point.  It seems like these keys and values might add to ambiguity around how to configure.  Perhaps I don't see how we might benefit from having them.  I would love to hear @Wangda's perspective.  If there isn't a compelling reason to have these extra keys, cleaning them up would be ideal.

File: rlclientlib/azure_factories.cc [hunk 1]
```diff
@@ -98,6 +101,17 @@ int create_apim_http_api_sender(i_sender** retval, const u::configuration& cfg,
   return error_code::success;
 }
 
+// Creates i_sender object for sending episode data to the apim endpoint.
+int episode_api_sender_create(i_sender** retval, const u::configuration& cfg, error_callback_fn* error_cb,
+    i_trace* trace_logger, api_status* status)
+{
+  const auto* const api_host = cfg.get(name::EPISODE_HTTP_API_HOST, "localhost:8080");
+  return create_apim_http_api_sender(retval, cfg, api_host, cfg.get_int(name::EPISODE_EH_MAX_HTTP_RETRIES, 4),
```
[the remark this answers is not in the document]
Comment (homezcx) [reply rlclientlib/azure_factories.cc]: Thanks for checking. Fixed.

File: include/constants.h [hunk 2]
```diff
@@ -34,10 +34,14 @@ const char* const EPISODE_EH_HOST = "episode.eventhub.host";
 const char* const EPISODE_EH_NAME = "episode.eventhub.name";
 const char* const EPISODE_EH_KEY_NAME = "episode.eventhub.keyname";
 const char* const EPISODE_EH_KEY = "episode.eventhub.key";
+const char* const EPISODE_HTTP_API_HOST = "episode.http.api.host";
```
[the remark this answers is not in the document]
Comment (homezcx) [reply include/constants.h]: Thanks for checking. Fixed.

File: rlclientlib/azure_factories.cc [hunk 3]
```diff
@@ -98,6 +101,17 @@ int create_apim_http_api_sender(i_sender** retval, const u::configuration& cfg,
...
+      cfg.get_int(name::EPISODE_EH_TASKS_LIMIT, 4),
```
[the remark this answers is not in the document]
Comment (homezcx) [reply rlclientlib/azure_factories.cc]: Thanks for checking. Fixed.

File: rlclientlib/azure_factories.cc [hunk 4]
```diff
@@ -98,6 +101,17 @@ int create_apim_http_api_sender(i_sender** retval, const u::configuration& cfg,
...
+      std::chrono::milliseconds(cfg.get_int(name::EPISODE_EH_MAX_HTTP_RETRY_DURATION_MS, 3600000)), error_cb,
```
[the remark this answers is not in the document]
Comment (homezcx) [reply rlclientlib/azure_factories.cc]: Thanks for checking. Fixed.

**Review by rajan-chari — APPROVED** [review]

**Pull request merged**